### ISSUE TITLE
Add XML comment of Property to 'With' Method

### ISSUE THREAD

**Issue by lucavoit** [MelGrubb/BuilderGenerator#48]
It would be great if the comment of the generated With Method would be the same or atleast include any existing xml documentation of the property it is setting in the base class.

If you want I can help with implementing this

**Comment by MelGrubb** [comment]
It's an interesting idea. It should be possible to get from the semantic model back to the documentation. Getting it to play nicely with the internal caching mechanism would be tricky, though. Basically, the comments would have to become part of the BuilderInfo object if you want changes in the comments to trigger a regeneration of the builder. Now you've become responsible for digging the comments out of the syntax tree in the transform method at which point you're taking on the responsibility for stitching together the information across potentially multiple partial classes. Oh, and the comment might be a single line or split across multiple. If you can live with the comments only updating when some _other_, more significant change is detected, then that becomes easier.

tl;dr - Possible, but only changes to properties in the BuilderInfo instance returned from the Transform method will trigger regeneration, and extracting comments at the syntax tree level sounds like no fun.

**Comment by lucavoit** [comment]
I had a quick look into this before creating this issue and I came up with a POC (Dotnetfiddle: https://dotnetfiddle.net/zNdhig).
The interesting part there is the GetDocumentationCommentXml method of the IPropertySymbol which you also use in BuilderGenerator:208 Transform method. this method supports multiline comments as you can see in the fiddle.

If you want I can open a PR with the proposed implementation, or I can give you the changes I tried some other way

**Comment by MelGrubb** [comment]
A PR would always be welcome. I'm making a big push for version 3, addressing some long-standing issues, and generally cleaning things up. If you already have a direction for this, then I'd love to see it. I'll probably take a swing at it too, but there are a number of issues I'm chasing, so I can't say how long before I can get this implemented.

**Comment by MelGrubb** [comment]
Closed by #57
